.github/workflows/docker.yml

```diff
@@ -0,0 +1,142 @@
+name: Docker Test
+on:
+  pull_request:
+  push:
+    branches: [ main ]
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  docker:
+    name:  ${{ matrix.os }}-${{ matrix.version }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        os:
+          - ubuntu
+          - debian
+          - centos
+        build_type:
+          #- Debug
+          - Release
+        compiler:
+          - gcc
+          #- clang
+        include:
```

Comment:
Hmm. I thought I would see a bazillion combos on the CI run but there are only 3 from this workflow. Is that expected?

Reply:
Yeah, I noticed this too but had to step out. I'm trying to figure out what's going on now.